pyresample/resampler.py

```diff
@@ -132,3 +148,193 @@ def _create_cache_filename(self, cache_dir=None, prefix='',
+
+    Resample_blocks applies a function blockwise to transform data from a source
+    area domain to a destination area domain.
+
+    Args:
+        funk: A callable to apply on the input data. This function is passed a block of src_arrays,
+            dst_array in that order, followed by the kwargs, which include the fill_value. If the function has a
+            block_info keyword argument, block information is passed to it that provides the source area, destination
+            area, position of source and destination blocks relative to respectively src_area and dst_area.
+        src_area: a source geo definition.
+        dst_area: a destination geo definition. If the same as the source definition, a ValueError is raised.
+        funk: a function to use. If func has a block_info keyword argument, the chunk info is passed, as in map_blocks
+        src_arrays: data to use. When split into smaller bit to pass to funk, they are split across the x and y
+            dimensions, but not across the other dimensions, so all the dimensions of the smaller arrays will be using
+            only one chunk!
+        dst_arrays: arrays to use that are already in dst_area space. If the array has more than 2 dimensions,
+            the last two are expected to be y, x.
+        chunk_size: the chunks size(s) to use in the dst_area space. This has to be provided since it is not guaranteed
+            that we can get this information from the other arguments. Moreover, this needs to be an iterable of k
+            elements if the resulting array of funk is to have a different number of dimensions than the input array.
+        dtype: the dtype the resulting array is going to have. Has to be provided.
+        kwargs: any other keyword arguments that will be passed on to funk.
+
+
+    Principle of operations:
+        Resample_blocks works by iterating over chunks on the dst_area domain. For each chunk, the corresponding slice
+        of the src_area domain is computed and the input src_arrays are cut accordingly to pass to funk. To know more
+        about how the slicing is performed, refer to the :class:Slicer class and subclasses.
+
+    Examples:
+        To generate indices from the gradient resampler, you can apply the corresponding function with no input. Note
+        how we provide the chunk sizes knowing that the result array with have 2 elements along a third dimension.
+
+        >>> indices_xy = resample_blocks(gradient_resampler_indices, source_geo_def, [], target_geo_def,
+        ...                              chunk_size=(2, "auto", "auto"), dtype=float)
+
+        From these indices, to resample an array using bilinear interpolation:
+
+        >>>  resampled = resample_blocks(block_bilinear_interpolator, source_geo_def, [src_array], target_geo_def,
+        ...                              dst_arrays=[indices_xy],
+        ...                              chunk_size=("auto", "auto"), dtype=src_array.dtype)
+
+
+    """
```

Comment:
Can you triple check that this renders well in the HTML docs?

Reply:
Neither gradient nor EWA resampler apis are being shown...
https://pyresample--341.org.readthedocs.build/en/341/api/pyresample.ewa.html#module-contents

Reply:
looks like it's because the cython parts are not built and thus fail to import. @djhoese have fixed something like this before?

Reply:
I also see this in the build log:

```
WARNING: autodoc: failed to import module 'resampler' from module 'pyresample'; the following exception was raised:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyresample/conda/341/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 70, in import_module
    return importlib.import_module(modname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyresample/conda/341/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyresample/checkouts/341/pyresample/resampler.py", line 307, in <module>
    @lru_cache
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyresample/conda/341/lib/python3.7/functools.py", line 490, in lru_cache
    raise TypeError('Expected maxsize to be an integer or None')
TypeError: Expected maxsize to be an integer or None
```

Another thing is that I need to update the docs to automatically generate the API files when the site is generated. And...I don't see anything that should suggest the cython extensions shouldn't be built. Cython is in the build environment and `conf.py` isn't mocking anything. Are you able to generate the docs locally?

Reply:
This seems not to happen with Python 3.9, but does happen with 3.7.

Reply:
From https://stackoverflow.com/questions/47218313/use-functools-lru-cache-without-specifying-maxsize-parameter it seems that with Python 3.7 one needs to use `@lru_cache()` instead of `@lru_cache`. This also works at least with Python 3.9.

Reply:
I think it looks like it renders decently well:

![image](https://user-images.githubusercontent.com/1828519/171678490-37a5ec88-a78d-4336-8739-8eccedee16ce.png)

Reply:
Now that the doc builds fine (thanks Dave), I could check the formatting and it looks fine!